pandas/core/indexing.py

```diff
@@ -88,8 +89,8 @@ class IndexingError(Exception):
 
 
 class _NDFrameIndexer(_NDFrameIndexerBase):
-    _valid_types = None
-    _exception = KeyError
+    _valid_types = None  # type: Optional[str]
```

Comment:
Is this actually optional beyond declaration?

Reply:
No, not really.